### ISSUE TITLE
time.Time has different precision depending on the operating system

### ISSUE THREAD

**Issue by davidterranova** [golang/go#41087]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13.12 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

<pre>
$ go version
go version go1.15 darwin/amd64
$ go run main.go
2020-08-28 10:37:50.388561 +0200 CEST m=+0.000120847
</pre>

<pre>
$ go version
go version go1.15 linux/amd64
$ go run main.go
2020-08-28 08:44:47.598714765 +0000 UTC m=+0.000071254
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/davidterranova/Library/Caches/go-build"
GOENV="/Users/davidterranova/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/davidterranova/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/opt/go/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/opt/go/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/davidterranova/Documents/development/smallpdf/pro/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/j_/6_9hdtls6tb_yvmfygpy54jm0000gn/T/go-build886013447=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

```go
package main

import (
  "time""
  "fmt"
)

func main() {
  fmt.Println(time.Now())
}
```
Has different time precisions depending on the operating system it is being ran.
- micro second precision on MacOS Catalina
- nano second precision on linux
- micro second on Windows

### What did you expect to see?

Either the same time precision or the mention of these differences in the documentation of the package `time`

**Comment by davecheney** [comment]
What about time stamps received from file systems? The reason I ask is different file systems support different levels of precision so the fidelity of an os.FileInfo is dependant on which file system the file is referenced from. 

Istm that in both the time now case and the os fileinfo case the only thing that can be documented is the precision of the time stamp value is operating system dependant.

**Comment by alexbrainman** [comment]
> * micro second on Windows

How did you measure that? Because it is about 1 millisecond, if you use time.Sleep function. I agree with @davecheney - it is too complicated to explain. And what happens, if things will change in time? The only thing we can document that it is `operating system dependant`. I just don't see how that will help anyone.

Alex

**Comment by davidterranova** [comment]
@davecheney Filesystems are IO. I don't think we can do much. Same for databases (postgreSQL does support micro second precision only).

@alexbrainman I didn't but peterGo on the mailing list mentioned it.

I think we can improve the documentation mentioning the time precision can be different depending on the os

**Comment by bcmills** [comment]
I don't think it is the precision of `time.Time` itself that actually varies. Rather, it is the precision of `time.Now()`. (To my knowledge, timestamps computed with `time.UnixNano` and `(time.Time).Add` have the same precision on all platforms.)
